pkg/reconciler/intevents/pullsubscription/reconciler.go

```diff
@@ -309,6 +310,17 @@ func (r *Base) reconcileDataPlaneResources(ctx context.Context, ps *v1.PullSubsc
 		logging.FromContext(ctx).Desugar().Error("Error serializing tracing config", zap.Error(err))
 	}
 
+	authType, err := authtype.GetAuthType(ctx, r.ServiceAccountLister, nil, authtype.AuthTypeArgs{
```

Comment:
The controller is supposed to only have permission to watch/list/get secret in control plane's namespace (https://github.com/google/knative-gcp/issues/1827), so when the source is using secret mode, it will directly return authType as "secret" without further check, and the secret related error will be handled in Step3 in the design (Additional k8s Event check for Secret absence). The secret lister is not used here for sources. 

Previously, I use the same GetAuthType function for BrokerCell and Sources, it needs to pass the secret lister. Following your later comment, I split the function. Now, for sources, getAuthType only asks to pass serviceaccount lister in.

Reply:
Thanks for the detailed explanation. I think the change you made makes sense.